File: 3291-find-if-array-can-be-sorted/3291-find-if-array-can-be-sorted.py
class Solution:
    def canSortArray(self, nums: List[int]) -> bool:
        l = 0
        h = nums[0]
        bit = False
        for i in range(1, len(nums)):
            bit = (bin(h).count('1') != bin(nums[i]).count('1'))
            if bit:
                l = h
            if nums[i] > h:
                h = nums[i]
            if nums[i] < l:
                return False
        
        return True

File: 3291-find-if-array-can-be-sorted/3291-find-if-array-can-be-sorted.cs
public class Solution {
    public int count(int n) {
        int b = 0;
        while (n != 0) {
            n = n & (n - 1);
            b++;
        }
        return b;
    }
    public bool CanSortArray(int[] nums) {
        var l = 0;
        var h = nums[0];
        bool bit;
        for (var i = 1; i < nums.Length; ++i) {
            bit = (count(h) != count(nums[i]));
            if (bit) 
                l = h;
            if (nums[i] > h)
                h = nums[i];
            if (nums[i] < l)
                return false;
        }
        return true;
    }
}